src/core/sockopts.pm6

```diff
@@ -0,0 +1,22 @@
+my role SockOptLike {
+    multi method CALL-ME(Int() $sonum) {
+        return self unless $sonum;
+        nextsame
+    }
+}
+
+my enum SockOpt does SockOptLike (
+    |nqp::stmts(
+        ( my \res  := nqp::list ),
+        ( my \iter := nqp::iterator(nqp::getsockopts()) ),
+        nqp::while(
+            iter,
+            nqp::stmts(
+                ( my \p := nqp::p6bindattrinvres(nqp::create(Pair), Pair, '$!key', nqp::shift(iter)) ),
+                nqp::bindattr(p, Pair, '$!value', nqp::abs_i(nqp::shift(iter)) ),
+                nqp::push(res, p)
+            )
+        ),
+        res
+    )
+);
```

Comment:
One thing to note is that in the enum definition, a separate lexical scope isn't being created, so the variables introduced there leak into the [CORE lexical environment](https://github.com/rakudo/rakudo/issues/2773). A mistake I'm guilty of as well 😝.

Check out  @lizmat's [commit here](https://github.com/rakudo/rakudo/commit/76b6416d00dca8dc2f669b438581e34fefdf49bb) which adds a new lexical scope and avoids leakage.